### PR TITLE
add inception v4

### PR DESCRIPTION
**Pull request by makaveli10**
Sanity Checks done:

- create model and print def.
- generate tensorrt weights file.

### CODE REVIEW

File: models/inception_v4/model.py [hunk 1]
```diff
@@ -0,0 +1,385 @@
+""" Pytorch Inception-V4 implementation
+Sourced from https://github.com/Cadene/pretrained-models.pytorch (MIT License)
+"""
+
+from __future__ import print_function, division, absolute_import
```
Comment (shivamswarnkar) [inline models/inception_v4/model.py]: We are not giving py2.7 support in other files, so do you think we should give it here?

File: models/inception_v4/model.py [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+        x = self.last_linear(x)
+        return x
+
+    def forward(self, input):
+        x = self.features(input)
+        x = self.logits(x)
+        return x
+
+
+class InceptionV4:
```
Comment (shivamswarnkar) [inline models/inception_v4/model.py:8]: Would it be a good idea to move this class into a different file to keep the overall structure consistent? In other words, I think we should break this file into two parts.

File: models/inception_v4/model.py [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+    """Initialize Inceptionv4 model.
```
Comment (shivamswarnkar) [inline models/inception_v4/model.py]: I think we should move this docstring into init. And instead add a class level doc here (keeping docstrings consistent as other model classes)

File: models/inception_v4/model.py [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+            nn.Module: torch model
+        """
+        return self._model
+
+    def print_summary(self) -> None:
+        """Print summary of the model.
+        """
+        print(summary(self._model, input_size=(3, 299, 299)))
+
+    def load_model(self, num_classes, pretrained) -> None:
```
Comment (shivamswarnkar) [inline models/inception_v4/model.py]: - Do you want users to call this method after object creation? If not, then would it be a better idea to make this method protected/private? 
- We should add type info to parameters (ex num_classes:int, etc.)

File: models/inception_v4/model.py [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.utils.model_zoo as model_zoo
+
+from torchsummary import summary
+
+
+pretrained_settings = {
```
Comment (shivamswarnkar) [inline models/inception_v4/model.py]: should we make it a JSON config like our other projects?

File: models/inception_v4/model.py [hunk 6]
```diff
@@ -0,0 +1,385 @@
...
+            'input_range': [0, 1],
+            'mean': [0.5, 0.5, 0.5],
+            'std': [0.5, 0.5, 0.5],
+            'num_classes': 1001
+        }
+    }
+}
+
+
+class BasicConv2d(nn.Module):
```
Comment (shivamswarnkar) [inline models/inception_v4/model.py]: Let's add some #Todo here to update the docstrings later on.

**Review by shivamswarnkar — CHANGES_REQUESTED** [review]
Please take a look at the comments I have added, and let me know your thoughts.

File: models/inception_v4/model.py (hunk 1 above)
Comment (makaveli10) [reply models/inception_v4/model.py:5]: No we shouldn't.

File: models/inception_v4/model.py (hunk 2 above)
Comment (makaveli10) [reply models/inception_v4/model.py:8]: Sounds good.

File: models/inception_v4/model.py (hunk 1 above)
Comment (makaveli10) [reply models/inception_v4/model.py]: No i dont think so. removed it

File: models/inception_v4/model.py (hunk 2 above)
Comment (makaveli10) [reply models/inception_v4/model.py:8]: Done.

File: models/inception_v4/model.py (hunk 3 above)
Comment (makaveli10) [reply models/inception_v4/model.py]: Done.

**Review by shivamswarnkar — COMMENTED** [review]
LGTM

**Pull request merged**